doc/specification/language-proposal.md

```diff
@@ -0,0 +1,1518 @@
+>
+> _module-import_ ::= `import` ***symbol*** ( `as` ***symbol*** )
+>
+> _type-alias_ ::= `type` ***symbol*** `=` _type-expr_ `;`
+>
+> _parameter-defn_ ::= `parameter` ***symbol*** _type-assertion_? `=` _expression_ `;`
+>
+> _constant-defn_ ::= `def` ***symbol*** _type-assertion_? `=` _expression_ `;`
+>
+> _function-defn_ ::= `def` ***symbol*** `=` _function-literal_ `;`
+
+#### Module import
+<a id="module-imports"/>
+
+Identifiers bound in module scope in one module can be used in another module or interface, if the module is imported. If a module `A` is imported as `B`, a type, parameter, constant, or function _x_ defined in `A` can be referenced with the qualified identifier `B`._x_.
+
+Examples:
+```
+module A {
+    constant real pi = 3;
+}
+
+module X {
+    import module A;
+    constant real pi = A.pi; # Also 3.
+}
+
+module Y {
+    import module A as B;
+    constant real pi = B.pi; # Also 3.
+}
+```
+
+Note that in a module import `import module A as B`, `A` is interpreted in module context, while `B` is interpreted in expression context. So the following is well-formed,
+```
+module A { }
+module B {
+   def A = 3;
+   import A as M;
+}
+```
+but
+```
+module A { }
+module B {
+   def M = 3;
+   import A as M;
+}
+```
+is not, as `M` is being rebound in module scope in an expression context.
+
+
+#### Type aliases
+<a id="type-aliases"/>
+
+A type alias binds a non-function type to an identifier in module scope, in type contexts. In addition, if the bound type is a record or quantity type, the derivative type of the bound type is bound to the identifier suffixed with an apostrophe `'` (or equivalently, any of the accepted prime marks).
+
+Example:
+```
+type foo = metre;
+type bar = { a: time; b: foo };
+
+# type aliases used to assert type in constant definitions:
+def r: foo = 10 m;
+def s: bar = { a = 3 s; b = 20 m; };
+
+# derivative type aliases are bound implicitly:
+def r': foo' = 1 m/s;
+def s': bar' = { a' = 1.2; b' = 3.4 m/s; };
+
+```
+
+#### Parameters
+
+A parameter definition introduces a new identifier in module scope, in expression contexts, together with a default value given by a _paramater-constant_ expression — that is, the expression can depend only upon constants and other parameters (see [Expressions](#expressions)).
+
+Parameters that are exported in an interface can be bound to a user supplied value externally; they are otherwise constant. In any expression in the same module scope that uses that parameter, the value of the parameter will be the user supplied value. This applies to expressions that are bound to other parameters — for example, consider the following module and interface definition.
+
+```
+module impl {
+    parameter a: voltage = 3 mV;
+    parameter I: current = (13 mV - a) / 20 kΩ;
+}
+
+interface point "foo" {
+    import impl;
+    export parameter impl.a as a;
+
+    effect current = impl.I;
```

Comment:
We implicitly import qualified, can we have the equivalent of Python's `from M import *`?
Also, I wonder how imported types look like
```
import CGS;

[...]
let E = 42 CGS.erg
```
like this?

Reply:
And finally, can we hide items when importing and/or exporting? Especially useful when pulling in all things into the global scope.

Reply:
The motivation for not having a `from M import *` is to avoid any non-obvious name collisions in module scope. Perhaps this could be a convenience we add later if there is demand and/or frustration?

Without custom units, something like `42 CGS.erg` will not work. `erg` would have to be a constant:
```
module CGS {
    def erg = 1e-7 J;
}

module A {
    import CGS;
    # ...
    def E = 42  * CGS.erg;
}

# or
module B {
    import CGS
    def erg = CGS.erg;
    # ...
    def E = 42 * erg;
 }